File: web/src/models/response.ts
export interface ValueResponse {
  status: number;
  value: unknown;
}

export interface ErrorResponse {
  status: number;
  error: string;
}

export interface SessionResponse {
  status: number;
  sessionId: string;
}

export interface FileUploadResponse {
  status: number;
  file: string;
}

export interface TaskStateResponse {
  status: number;
  task: string;
  taskState: string;
}

File: api/models/response.go
package models

type ValueResponse struct {
	Status int `json:"status"`
	Value  any `json:"value"`
}

type ErrorResponse struct {
	Status int    `json:"status"`
	Error  string `json:"error"`
}

type SessionResponse struct {
	Status    int    `json:"status"`
	SessionId string `json:"sessionId"`
}

type FileUploadResponse struct {
	Status int    `json:"status"`
	File   string `json:"file"`
}

type TaskStateResponse struct {
	Status    int    `json:"status"`
	Task      string `json:"task"`
	TaskState string `json:"taskState"`
}

type SummaryResponse struct {
	Status  int    `json:"status"`
	Summary string `json:"summary"`
}
